doc/reference/storage/fcb/fcb.rst

```diff
@@ -0,0 +1,74 @@
+.. _fcb-manual:
+
+Flash Circular Buffer (FCB)
+###########################
+
+Flash circular buffer provides an abstraction through which you can treat
+flash like a FIFO. You append entries to the end, and read data from the
+beginning.
+
+Description
+***********
+
+Entries in the flash contain the length of the entry, the data within
+the entry, and checksum over the entry contents.
+
+Storage of entries in flash is done in a FIFO fashion. When you
+request space for the next entry, space is located at the end of the
+used area. When you start reading, the first entry served is the
+oldest entry in flash.
+
+Entries can be appended to the end of the area until storage space is
+exhausted. You have control over what happens next; either erase oldest
+block of data, thereby freeing up some space, or stop writing new data
+until existing data has been collected. FCB treats underlying storage as
+an array of flash sectors; when it erases old data, it does this a
+sector at a time.
+
+Entries in the flash are checksummed. That is how FCB detects whether
+writing entry to flash completed ok. It will skip over entries which
+don't have a valid checksum.
+
+Usage
+*****
+
+To add an entry to circular buffer:
+
+- Call ``fcb_append()`` to get the location where data can be written. If
```

Comment:
Not required, but you can make all these function mentions into links to their API documentation by writing:
```
   ``fcb_append()``
```
as:
```
   :cpp:func:`fcb_append()`
```
or simply (with single back-ticks and no parentheses:
```
   `fcb_append`
```